pkg/local_object_storage/writecache/flush.go

```diff
@@ -270,9 +270,27 @@ func (c *cache) flushObject(obj *object.Object, data []byte) error {
 
 	_, err = c.metabase.UpdateStorageID(updPrm)
 	if err != nil {
+		if errors.Is(err, apistatus.ErrObjectNotFound) {
+			// we have the object and we just successfully put it so all the
+			// information for restoring the object is here; meta can be
+			// currupted, resynced, etc, just trying our best
+
+			var prmMeta meta.PutPrm
+			prmMeta.SetObject(obj)
+			prmMeta.SetStorageID(res.StorageID)
+
+			_, err = c.metabase.Put(prmMeta)
+			if err != nil {
+				err = fmt.Errorf("trying to restore missing object in metabase: %w", err)
+			}
+
+			return err
```

Comment:
Still makes me wonder _how_ this can happen. Some log is also good to have (failed to update but restored).

Reply:
any metabase resync with objects that are in WC but not in blobstor yet will show such behavior; also objects PUT in degraded mode will have the same problem. anyway, suggested code is correct: we want to update storage ID but we do not have object, so why not put it to metabase?